Documentation/user-guides/configuring-cluster-monitoring.md

```diff
@@ -1,137 +1,629 @@
+Monitoring a platform such as OpenShift requires a coordination of multiple components that must work well between themselves.
+However, users should be able to customize the monitoring stack in such a way that they end up with a resilient and highly available monitoring solution.
+Despite this, to avoid users from misconfiguring the monitoring stack of their clusters not all configuration parameters are exposed.
 
-This is an example configuration with all image parameters set to a custom registry:
+Configuring Cluster Monitoring is optional. If the config does not exist or is empty or malformed, then defaults will be used.
+
+## Configuration of OpenShift Container Platform
+
+The ConfigMap `cluster-monitoring-config` in the `openshift-monitoring` namespace is the root for all configuration of the core infrastruture monitoring. Here we can configure the following:
+
+```yaml
+# Holds configuration for the Alertmanager component in the cluster
+# from configuring nodeSelectors to enabeling dedicated alertmanager
+# instance for user-defined projects.
+alertmanagerMain: <struct>
+# Boolean flag to enable monitoring for user-defined projects.
+enableUserWorkload: <bool>
+# Holds configuration related with prometheus-adapter, nodeSelector and tolerations
+# and Audit level
+k8sPrometheusAdapter: <struct>
+# Holds configuration related with kube-state-metrics agent, nodeSelector and tolerations
+kubeStateMetrics: <struct>
+# Holds configuration related with the prometheus component that will be managed by
+# the prometheus-operator from configuring nodeSelector to RemoteWrite
+prometheusK8s: <struct>
+# Holds configuration related with prometheus-operator, nodeSelector, tolerations
+# and logLevel
+prometheusOperator: <struct>
+# Holds configuration related with openshift-state-metrics agent, nodeSelector and tolerations
+openshiftStateMetrics: <struct>
+# Holds configuration related with the Thanos Querier component, nodeSelector, tolerations
+thanosQuerier: <struct>
+```
+
+### alertmanagerMain
+
+Holds configuration for all things Alertmanager
 
-[embedmd]:# (../../examples/user-guides/configuring-cluster-monitoring/custom-image-config.yaml)
 ```yaml
-prometheusOperator:
-  baseImage: custom-registry.com/prometheus-operator
-  prometheusConfigReloaderBaseImage: custom-registry.com/prometheus-config-reloader
-  configReloaderBaseImage: custom-registry.com/configmap-reload
-prometheusK8s:
-  baseImage: custom-registry.com/prometheus
 alertmanagerMain:
-  baseImage: custom-registry.com/alertmanager
-auth:
-  baseImage: custom-registry.com/openshift-oauth-proxy
-nodeExporter:
-  baseImage: custom-registry.com/node-exporter
+  # Boolean flag to enable or disable the alertmanager instance that
+  # will live under openshift-monitoring
+  enabled: <bool>
+  # Boolean flag to enable or disable the alertmanager instance for
+  # user workload monitoring  under openshift-user-workload-monitoring
```

Comment:
this enables the reconciliation of `AlertmanagerConfig` resources but not the dedicated UWM alertmanager.